test/compressor_test.go

```diff
@@ -39,6 +40,78 @@ import (
 	testpb "google.golang.org/grpc/interop/grpc_testing"
 )
 
+// TestCompressionCases validates gRPC status codes for different client-server compression setups
+// ensuring the correct behavior when compression is enabled or disabled on either side.
+func (s) TestCompressionCases(t *testing.T) {
+	tests := []struct {
+		desc           string
+		clientUseNop   bool
+		serverUseNop   bool
+		expectedStatus codes.Code
+	}{
+		{
+			desc:           "Client and Server use nop compression",
+			clientUseNop:   true,
+			serverUseNop:   true,
+			expectedStatus: codes.OK,
+		},
+		{
+			desc:           "Only Client use nop compression",
+			clientUseNop:   true,
+			serverUseNop:   false,
+			expectedStatus: codes.Unimplemented,
+		},
+		{
+			desc:           "Only Server use nop compression",
+			clientUseNop:   false,
+			serverUseNop:   true,
+			expectedStatus: codes.Internal,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ss := &stubserver.StubServer{
+				UnaryCallF: func(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+					return &testpb.SimpleResponse{
+						Payload: in.Payload,
+					}, nil
+				},
+			}
+			sopts := []grpc.ServerOption{}
+			if test.serverUseNop {
+				sopts = append(sopts, grpc.RPCCompressor(newNopCompressor()), grpc.RPCDecompressor(newNopDecompressor()))
```

Comment:
Please add a comment mentioning why the deprecated server option is required. Same for the client.